v0/sources/mailmodo/transform.js

```diff
@@ -0,0 +1,92 @@
+const path = require("path");
+const fs = require("fs");
+const sha256 = require("sha256");
+const {
+  CustomError,
+  flattenJson,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+const Message = require("../message");
+
+// import mapping json using JSON.parse to preserve object key order
+const mapping = JSON.parse(
+  fs.readFileSync(path.resolve(__dirname, "./mapping.json"), "utf-8")
+);
+
+function settingProperties(event, message) {
+  const flatEvent = event;
+  delete flatEvent?.recipientData;
+  delete flatEvent?.recordedAt;
+  delete flatEvent?.triggerData?.formSubmissionData?.recordedAt;
+  delete flatEvent?.triggerData?.formSubmissionData?.recipientData;
+  message.properties = removeUndefinedAndNullAndEmptyValues(
+    flattenJson(flatEvent)
+  );
+
+  // deleting already mapped fields
+  delete message?.properties["triggerData.email"];
+  delete message?.properties["triggerData.userId"];
+  delete message?.properties["recipientData.email"];
+  delete message?.properties?.recipientEmail;
+  delete message?.properties["triggerData.formSubmissionData.email"];
+  delete message?.properties["triggerData.formSubmissionData.recipientEmail"];
```

Comment:
```suggestion
  message.properties = removeUndefinedAndNullAndEmptyValues(
    flattenJson(flatEvent)
  );
  // deleting already mapped fields
  delete message?.properties.recipientData;
  delete message?.properties.recordedAt;
  delete message?.properties["triggerData.formSubmissionData.recordedAt"];
  delete message?.properties["triggerData.formSubmissionData.recipientData"];
  delete message?.properties["triggerData.email"];
  delete message?.properties["triggerData.userId"];
  delete message?.properties["recipientData.email"];
  delete message?.properties?.recipientEmail;
  delete message?.properties["triggerData.formSubmissionData.email"];
  delete message?.properties["triggerData.formSubmissionData.recipientEmail"];
```
We can put these element inside an array and iterate through that to delete one by one.

Reply:
Done